### DISCUSSION TITLE
Is it invalid to call `Attribute` with the same key on the same node/way/relation?

### DISCUSSION THREAD

**Discussion by cldellow** [systemed/tilemaker#579] [General]
For example, is this valid, and is the expected result that last-one-in wins?

```lua
function node_function(node)
  node:Layer("somelayer")
  node:Attribute('name', 'Unknown')

  if ...
    node:Attribute('name', 'an actual name')
  end
end
```

I think neither the OMT process.lua nor Shortbread's process.lua do this currently.

Poking about in AttributeStore, it doesn't contemplate that a user would do this, and stores both name pairs. At writing time, I think both get written out in the order of `std::map`'s comparator, which results in the lexicographically biggest value winning -- in the above example, `Unknown` would be written.

I ask because I'm looking at making some changes to AttributeStore and am wondering if I need to preserve this behaviour, or if anything's fair game so long as it doesn't crash.

**Comment by systemed** [comment]
I would expect the last one to win - it's essentially what you'd expect if you were to write (in pseudocode)

    let a = 5
    let a = 3
    print(a)

which suggests we're not doing it right at present!

AttributeStore is quite extensively rewritten in #499 so that's probably the code to target.

**Comment by cldellow** [comment]
Great, thanks!
